### PR TITLE
server: ensure that service-defaults meta is incorporated into the discovery chain response

### PR DESCRIPTION
**Pull request by rboyer**
Also add a new "Default" field to the discovery chain response to clients (cc @johncowen )

TODO:

- [x] update this https://www.consul.io/docs/connect/l7-traffic/discovery-chain

### CODE REVIEW

File: agent/consul/discovery_chain_endpoint.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ func (c *DiscoveryChain) Get(args *structs.DiscoveryChainRequest, reply *structs
 			reply.Index = index
 			reply.Chain = chain
 
-			if chain.IsDefault() {
```
Comment (rboyer) [inline agent/consul/discovery_chain_endpoint.go:96]: :thinking: Actually I think I have to pull this change out and backport it to 1.11 and 1.10

**Review by erichaberkorn — APPROVED** [review]
The service meta portion of this looks good to me 👍 I reviewed the other part too, but don't have enough context to do a proper review.

File: api/discovery_chain.go [hunk 2]
```diff
@@ -109,9 +109,17 @@ type CompiledDiscoveryChain struct {
 	// non-customized versions.
 	CustomizationHash string
 
+	// Default indicates if this discovery chain is based on no
```
Comment (mkeeler) [inline api/discovery_chain.go:112]: Should the API docs be updated to account for these changes.

**Review by mkeeler — COMMENTED** [review]
Code looks great. Just the two minor things.

File: agent/consul/discoverychain/compile.go [hunk 3]
```diff
@@ -327,14 +333,51 @@ func (c *compiler) compile() (*structs.CompiledDiscoveryChain, error) {
...
+//	resolver action being applied is redirection to another resolver that is
+//	default, so we double check the resolver matches the requested resolver.
+//
+// NOTE: "default chain" mostly means that this is compatible with how things
+// worked (roughly) in consul 1.5 pre-discovery chain, not that there are zero
+// config entries in play (like service-defaults).
+func (c *compiler) determineIfDefaultChain() bool {
+	if false {
+		return c.entries.IsChainEmpty()
+	}
```
[the remark this answers is not in the document]
Comment (rboyer) [reply agent/consul/discoverychain/compile.go]: Reverted refactor mostly that didn't go anywhere that worked.

File: api/discovery_chain.go (hunk 2 above)
Comment (rboyer) [reply api/discovery_chain.go:112]: IIRC we don't currently document the full disco chain response at all.

**Review by mkeeler — APPROVED** [review]

**Pull request merged**